cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,344 @@ def preferred_ntp_clients(self):
+
+            new_line = ":".join(new_list)
+
+            # Replace existing line for user with modified line
+            shadow_contents = shadow_contents.replace(original_line, new_line)
+            LOG.debug("Updating %s for user %s", update_type, name)
+            try:
+                util.write_file(
+                    shadow_file, shadow_contents, omode="w", preserve_mode=True
+                )
+            except IOError as e:
+                util.logexc(LOG, "Failed to update %s file", shadow_file)
+                raise e
+        else:
+            util.logexc(
+                LOG, "Failed to update %s for user %s", shadow_file, name
+            )
+
+    def lock_passwd(self, name):
+        """
+        Lock the password of a user, i.e., disable password logins
+        """
+
+        # Check whether Shadow's or Busybox's version of 'passwd'.
+        # If Shadow's 'passwd' is available then use the generic
+        # lock_passwd function from __init__.py instead.
+        (_out, err) = subp.subp(["passwd", "--help"])
+        if not err.startswith("BusyBox"):
+            return super().lock_passwd(name)
+
+        cmd = ["passwd", "-l", name]
+        # Busybox's 'passwd', unlike Shadow's 'passwd', errors
+        # if password is already locked:
+        #
+        #   "passwd: password for user2 is already locked"
+        #
+        # with exit code 1
+        try:
+            (_out, err) = subp.subp(cmd, rcs=[0, 1])
+            if re.search(r"is already locked", err):
+                return True
+        except Exception as e:
+            util.logexc(LOG, "Failed to disable password for user %s", name)
+            raise e
+
+    def expire_passwd(self, user):
+        # Check whether Shadow's or Busybox's version of 'passwd'.
+        # If Shadow's 'passwd' is available then use the generic
+        # expire_passwd function from __init__.py instead.
+        (_out, err) = subp.subp(["passwd", "--help"])
+        if not err.startswith("BusyBox"):
+            return super().expire_passwd(user)
+
+        # Busybox's 'passwd' does not provide an expire option
+        # so have to manipulate the shadow file directly.
+        shadow_contents = None
+        shadow_file = self.shadow_fn
+        try:
+            shadow_contents = util.load_text_file(shadow_file)
+        except FileNotFoundError as e:
+            LOG.warning("Failed to read %s file, file not found", shadow_file)
+            raise e
+
+        # Find the line in /etc/shadow for the user
+        original_line = None
+        for line in shadow_contents.splitlines():
+            current_user = line.split(":")[0]
+            if current_user == user:
+                LOG.debug("Found /etc/shadow line matching user %s", user)
+                original_line = line
+
+        if original_line is not None:
+            # Replace existing line for user with modified line
+            new_list = original_line.split(":")
```

Comment:
just use the new_line_parts var split from above

Reply:
Done